core/jobs/batch_jobs/exploration_migration_jobs.py

```diff
@@ -0,0 +1,316 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs for adding proto_size_in_bytes attribute to exploration."""
+
+from __future__ import annotations
+
+import logging
+
+from core import feconf
+from core.domain import caching_services
+from core.domain import exp_domain
+from core.domain import exp_fetchers
+from core.domain import exp_services
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+import result
+
+from typing import Iterable, Sequence, Tuple
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import base_models
+    from mypy_imports import datastore_services
+    from mypy_imports import exp_models
+
+(base_models, exp_models,) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.exploration
+])
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MigrateExplorationJob(base_jobs.JobBase):
+    """Job that add the proto_size_in_bytes attribute."""
```

Comment:
Do I need to write a different job so that all the exploration can have the new attribute ('proto_size_in_bytes') which I am introducing in this PR?

Reply:
No. You're not adding a new field to the `ExplorationModel`, you're adding a value into one of its fields. As this is a generic migration operation and this job is for doing that operation it should have a generic description. If someone else writes a new migration too they won't have to write a new job, they will just use this one.

Reply:
I am bit confused here, I had added `proto_size_in_bytes` in `ExplorationModel` now, and it's a new attribute that we are introducing to the exp_domain object. 
And the migration job helps us so that all the explorations can have `proto_size_in_bytes` attribute.